prboom2/src/p_pspr.c

```diff
@@ -365,6 +424,36 @@ dboolean P_CheckAmmo(player_t *player)
   return false;
 }
 
+//
+// P_SubtractAmmo
+// Subtracts ammo, w/compat handling. In mbf21, use
+// readyweapon's "ammopershot" field if it's explicitly
+// defined in dehacked; otherwise use the amount specified
+// by the codepointer instead (Doom behavior)
+//
+// [XA] NOTE: this function is for handling Doom's native
+// attack pointers; of note, the new A_ConsumeAmmo mbf21
+// codepointer does NOT call this function, since it doesn't
+// have to worry about any compatibility shenanigans.
+//
+
+void P_SubtractAmmo(struct player_s *player, int compat_amt)
```

Comment:
I'd like a better name for `compat_amt`. Maybe `vanilla_amount`?

Reply:
Works for me. :P

Reply:
All done.